php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+  }
+
+  public StatementTree INNER_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.INNER_STATEMENT).is(
+      b.firstOf(
+        FUNCTION_DECLARATION(),
+        CLASS_DECLARATION(),
+        TRAIT_DECLARATION(),
+        INTERFACE_DECLARATION(),
+        STATEMENT()));
+  }
+
+
+  public GlobalStatementTree GLOBAL_STATEMENT() {
+    return b.<GlobalStatementTree>nonterminal(PHPLexicalGrammar.GLOBAL_STATEMENT).is(
+      f.globalStatement(
+        b.token(PHPKeyword.GLOBAL),
+        GLOBAL_VAR(),
+        b.zeroOrMore(f.newTuple16(b.token(COMMA), GLOBAL_VAR())),
+        EOS()));
+  }
+
+  public VariableTree GLOBAL_VAR() {
+    return b.<VariableTree>nonterminal(PHPLexicalGrammar.GLOBAL_VAR).is(
+      f.globalVar(
+        b.zeroOrMore(b.token(PHPLexicalGrammar.VARIABLE_VARIABLE_DOLLAR)),
+        COMPOUND_VARIABLE()));
+  }
+
+  public UnsetVariableStatementTree UNSET_VARIABLE_STATEMENT() {
+    return b.<UnsetVariableStatementTree>nonterminal(PHPLexicalGrammar.UNSET_VARIABLE_STATEMENT).is(
+      f.unsetVariableStatement(
+        b.token(PHPKeyword.UNSET),
+        b.token(LPARENTHESIS),
+        MEMBER_EXPRESSION(),
+        b.zeroOrMore(f.newTuple14(b.token(COMMA), MEMBER_EXPRESSION())),
+        b.token(RPARENTHESIS),
+        EOS()));
+  }
+
+  public YieldStatementTree YIELD_STATEMENT() {
+    return b.<YieldStatementTree>nonterminal(PHPLexicalGrammar.YIELD_STATEMENT).is(
+      f.yieldStatement(YIELD_EXPRESSION(), EOS()));
+  }
+
+  public SwitchStatementTree SWITCH_STATEMENT() {
+    return b.<SwitchStatementTree>nonterminal(PHPLexicalGrammar.SWITCH_STATEMENT).is(
+      b.firstOf(
+        f.switchStatement(
+          b.token(PHPKeyword.SWITCH),
+          PARENTHESIZED_EXPRESSION(),
+          b.token(PHPPunctuator.LCURLYBRACE),
+          b.optional(b.token(PHPPunctuator.SEMICOLON)),
+          b.zeroOrMore(SWITCH_CASE_CLAUSE()),
+          b.token(PHPPunctuator.RCURLYBRACE)
+        ),
+        f.alternativeSwitchStatement(
+          b.token(PHPKeyword.SWITCH),
+          PARENTHESIZED_EXPRESSION(),
+          b.token(PHPPunctuator.COLON),
+          b.optional(b.token(PHPPunctuator.SEMICOLON)),
+          b.zeroOrMore(SWITCH_CASE_CLAUSE()),
+          b.token(PHPKeyword.ENDSWITCH),
+          EOS())));
+  }
+
+  public SwitchCaseClauseTree SWITCH_CASE_CLAUSE() {
+    return b.<SwitchCaseClauseTree>nonterminal(PHPLexicalGrammar.SWITCH_CASE_CLAUSE).is(
+      b.firstOf(
+        f.caseClause(
+          b.token(PHPKeyword.CASE),
+          EXPRESSION(),
+          b.firstOf(b.token(PHPPunctuator.COLON), b.token(PHPPunctuator.SEMICOLON)),
+          b.zeroOrMore(INNER_STATEMENT())
+        ),
+        f.defaultClause(
+          b.token(PHPKeyword.DEFAULT),
+          b.firstOf(b.token(PHPPunctuator.COLON), b.token(PHPPunctuator.SEMICOLON)),
+          b.zeroOrMore(INNER_STATEMENT()))));
+  }
+
+  public WhileStatementTree WHILE_STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)